src/com/ppm/integration/agilesdk/connector/agilecentral/RallyTimeSheetIntegration.java

```diff
@@ -80,31 +165,65 @@ protected synchronized String convertDate(Date date){
 
 		final Date startDate = timeSheet.getPeriodStartDate().toGregorianCalendar().getTime();
 		final Date endDate = timeSheet.getPeriodEndDate().toGregorianCalendar().getTime();
-
+		
+		System.out.println("items--"+values.get(Constants.KEY_REMOVE_ITEMS).toString());
 		for(final Workspace workspace : workspaces){
-
+			if(!values.get(Constants.KEY_WORKSPACE).isEmpty() && !workspace.getId().equals(values.get(Constants.KEY_WORKSPACE))){
+				continue;
+			}
+			
 			List<Project> projects = rallyClient.getProjects(workspace.getId());
-			for(final Project project : projects){
-
-				List<Iteration> iterations = rallyClient.getIterations(project.getId());
-				for(final Iteration iteration : iterations){	
-					
-					List<Revision> revisions = new ArrayList<Revision>();
-					List<HierarchicalRequirement> hierarchicalRequirements = iteration.getHierarchicalRequirement();
+			for(final Project project : projects){	
+				if(!values.get(Constants.KEY_PROJECT).isEmpty() && !project.getId().equals(values.get(Constants.KEY_PROJECT))){
+					continue;
+				}
+				
+				if(values.get(Constants.KEY_DATA_DETAIL_LEVEL).equals("1")){
+					List<HierarchicalRequirement> hierarchicalRequirements = rallyClient.getHierarchicalRequirements();
 					for(HierarchicalRequirement hierarchicalRequirement : hierarchicalRequirements){
+						if(!hierarchicalRequirement.getProjectUUID().equals(project.getUUID())){
+							continue;
+						}
+						if(hierarchicalRequirement.getChildrenCount() != 0){
+							continue;
+						}
 						RevisionHistory revisionHistory = rallyClient.getRevisionHistory(hierarchicalRequirement.getRevisionHistoryRef());
 						List<Revision> revision = rallyClient.getRevisions(revisionHistory.getId());
-						revisions.addAll(revision);							
+						HashMap<String, String> hms = getTimeSheetData(startDate, endDate, revision);
+						if(values.get(Constants.KEY_REMOVE_ITEMS).equals("true") && hms.size() == 0){
+							continue;
+						}
+						items.add(new RallyExternalWorkItem(subscription.getName(), workspace.getName(), project.getName(), 
+								hierarchicalRequirement.getName(),
+								hms,
+								values,
+								startDate,
+								endDate));
+					}					
+				}else{	
```

Comment:
Looks like your code is not correctly formatted. You should format automatically based on Eclipse formatting style files located in <PPM_ROOT>\Utilities\eclipse\codestyles.

The info on how to import this in Eclipse is on the wiki: https://rndwiki.corp.hpecorp.net/confluence/display/itg/PPM+Java+Code+Style